File: frontend/src/views/stores/repositories/storeRepository.ts
import ApiService from '@/services/ApiService'
import { Store, StoreProduct, CreateStoreDto, UpdateStoreDto, GetStoresQueryDto, GetStoreProductsQueryDto, StoresResponse, StoreProductsResponse } from '../types/store.types'

export class StoreRepository {
  static async getStores(query: GetStoresQueryDto = {}): Promise<StoresResponse> {
    return ApiService.fetchDataWithAxios<StoresResponse>({
      url: '/stores',
      method: 'get',
      params: query,
    })
  }

  static async getStore(id: string): Promise<Store> {
    return ApiService.fetchDataWithAxios<Store>({
      url: `/stores/${id}`,
      method: 'get',
    })
  }

  static async getStoreProducts(storeId: string, query: GetStoreProductsQueryDto = {}): Promise<StoreProductsResponse> {
    return ApiService.fetchDataWithAxios<StoreProductsResponse>({
      url: `/stores/${storeId}/products`,
      method: 'get',
      params: query,
    })
  }

  static async createStore(data: CreateStoreDto): Promise<Store> {
    return ApiService.fetchDataWithAxios<Store>({
      url: '/stores',
      method: 'post',
      data,
    })
  }

  static async updateStore(id: string, data: UpdateStoreDto): Promise<Store> {
    return ApiService.fetchDataWithAxios<Store>({
      url: `/stores/${id}`,
      method: 'patch',
      data,
    })
  }

  static async deleteStore(id: string): Promise<void> {
    return ApiService.fetchDataWithAxios<void>({
      url: `/stores/${id}`,
      method: 'delete',
    })
  }
}

File: choppi-app/choppiapp/lib/features/stores/data/repositories/store_repository.dart
import '../models/store_models.dart';
import '../services/store_service.dart';

class StoreRepository {
  final StoreService _storeService;

  StoreRepository(this._storeService);

  Future<StoresResponse> getStores({String? query, int page = 1, int limit = 10}) async {
    return await _storeService.getStores(query: query, page: page, limit: limit);
  }

  Future<Store> getStore(String id) async {
    return await _storeService.getStore(id);
  }

  Future<StoreProductsResponse> getStoreProducts(
    String storeId, {
    String? query,
    bool? inStock,
    int page = 1,
    int limit = 10,
  }) async {
    return await _storeService.getStoreProducts(
      storeId,
      query: query,
      inStock: inStock,
      page: page,
      limit: limit,
    );
  }
}